pkg/parser/mysql/util.go

```diff
@@ -86,10 +86,10 @@ var defaultLengthAndDecimalForCast = map[byte]lengthAndDecimal{
 
 // GetDefaultFieldLengthAndDecimalForCast returns the default display length (flen) and decimal length for casted column
 // when flen or decimal is not specified.
-func GetDefaultFieldLengthAndDecimalForCast(tp byte) (flen int, decimal int) {
+func GetDefaultFieldLengthAndDecimalForCast(tp byte) (flen, decimal int) {
 	val, ok := defaultLengthAndDecimalForCast[tp]
 	if ok {
-		return val.length, val.decimal
+		return int(val.length), val.decimal
```

Comment:
This will truncate the value. Does your application need to change the return value `flen` to int64?

Reply:
I had a couple of experiments with this, where I tried changing the return value to int64. But, this has a couple of issues:

* A change from `int` to `int64` has a domino-effect like effect, where the type changes needs to be propagated throughout many files. It also breaks tests.
* Ideally, the length of something should not be a signed integer, but perhaps `uint64`.
* For the code to be idiomatic Go, `-1` should not be used to indicate an error or a missing value, but an error should be returned instead. `-1` is currently used in several places, making an introduction of `uint64` a larger change.

So yes, ideally, uint, uint32 or uint64 should be used instead of int or int64 for lengths all over the code base, but this is a project of its own.

Reply:
My application does not need flen or int or int64 in particular, only that the code compiles and runs also on ARM7, ideally.